docs/databases/migrations.md

```diff
@@ -4,4 +4,568 @@ title: GraphQL Migrations
 sidebar_label: GraphQL Migrations
 ---
 
-## TODO
\ No newline at end of file
+Graphback uses [graphql-migrations](https://www.npmjs.com/package/graphql-migrations) to automatically create and update tables from a GraphQL schema.
+The package is able to calculate diff between your schema and your database to only executes required changes and thus keeping your database and schema in sync.
+
+## Compatibility
+
+- PostgreSQL (create and update database)
+- SQLite (create database only)
+
+## Usage
+
+Graphback migrations operates on business models defined in your schema: These are graphql types decorated with a [`@model`](../model/datamodel#Model) annotation. 
+The package exports an entry point which you can programatically setup in your source code and have it peform the migrations. 
+
+### API
+
+The `migrateDB` method creates and updates your tables and columns to match your GraphQL schema.
+All the database operations are wrapped in a single transaction, so your database will be fully rolled back to its initial state if an error occurs.
+The method takes three arguments as descripted in [migrations options](./migrations#Options) section.
+
+```ts
+import { migrateDB } from 'graphql-migrations';
+
+const schemaText = ```
+"""
+@model
+"""
+type Note {
+  id: ID!
+  title: String!
+}
+```;
+
+const dbConfig = {
+  // Knex.js based configuration
+};
+
+migrateDB(dbConfig, schemaText, {
+  // Additional options
+}).then((ops) => {
+    console.log(ops);
+});
+...
+```
+
+Assuming the above code is ran agains a PostgreSQL database, the following relations will be created:
+```sql
+\d
+              List of relations
+ Schema |    Name     |   Type   |   Owner    
+--------+-------------+----------+------------
+ public | note        | table    | postgresql
+ public | note_id_seq | sequence | postgresql
+```
+
+And the `note` table:
+
+```sql
+ \d note
+                                 Table "public.note"
+ Column |          Type          |                     Modifiers                     
+--------+------------------------+---------------------------------------------------
+ title  | character varying(255) | not null
+ id     | integer                | not null default nextval('note_id_seq'::regclass)
+Indexes:
+    "note_pkey" PRIMARY KEY, btree (id)
+```
+
+## Designing your data model
```

Comment:
I think a lot of this is duplicated from the Data Model page and can quickly go out of sync with each other. Though this does have only the migration specific information. Up for discussion what we do with this.

Reply:
I could not find a better to do this part. I have added links above and I think it is important to note that when modifying docs for a model / annotation etc one has to verify that the migration package is not broken and this includes the docs. The chance of the two docs being out of sync are there but very minimal considering that we do not go into details about what `@model` contains for example. Having said that, any suggestions welcomed about how we can re-restructure this part.

Reply:
What I meant by "going out of sync" was that because the same content needs to be duplicated in multiple places, the likelihood of one not getting updated by accident and becoming out of sync with the other is possible.

Though you are right, I think here there needs to be duplication, especially considering this library can be used without Graphback.